pkg/transport/custom_transform_collection.go

```diff
@@ -0,0 +1,279 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transport
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+
+	"github.com/kubestellar/kubestellar/api/control/v1alpha1"
+	controlclient "github.com/kubestellar/kubestellar/pkg/generated/clientset/versioned/typed/control/v1alpha1"
+	"github.com/kubestellar/kubestellar/pkg/jsonpath"
+)
+
+type customTransformCollection interface {
```

Comment:
The (optional) comment touching on using interfaces to hide dev-mistakes-error-prone fields that require delicate handling was referring to internals of the implementation.

This is ok too - but this interface lacks documentation (docs on this interface are what would appear to the developer during use in the transport code). Once fixed I'll LGTM.

Reply:
The force-push to c31f8e154 adds comment on the new internal interface.